pilot/pkg/xds/ads.go

```diff
@@ -1037,3 +1013,14 @@ func orderWatchedResources(resources map[string]*model.WatchedResource) []*model
 	}
 	return wr
 }
+
+// reportAllEvents is to simplify the logic of reporting events to the status reporter.
+// It reports all tracking events{CDS,EDS,LDS,RDS} for the given connection.
+func reportAllEvents(con *Connection, statusReporter DistributionStatusCache, nonce string) {
+	if statusReporter == nil {
+		return
+	}
+	for distributionType := range AllTrackingEventTypes {
```

Comment:
This is broken by removing ignoreEvents... this means we report we received it immediately. Wait will just ALWAYS return true.

The old logic was correct. If we push {CDS, EDS} we immediately report LDS,RDS as "ready" (logically they are, it was just a NOP). Once CDS/EDS is ACKed, we report those as ready respectively.

Reply:
But from your above comment https://github.com/istio/istio/pull/48542#discussion_r1462198197

The ack order is not determined. We should record all that state in one place. Rather than some in push, some in ack

Reply:
That doesn't give the behavior we are trying to express though.

Reply:
replied above